pkgs/java_http/lib/src/java_client.dart

```diff
@@ -39,23 +39,29 @@ class JavaClient extends BaseClient {
     // See https://github.com/dart-lang/http/pull/980#discussion_r1253700470.
     _initJVM();
 
-    final (statusCode, reasonPhrase, responseHeaders, responseBody) =
-        await Isolate.run(() {
-      request.finalize();
+    // We can't send a StreamedRequest to another Isolate.
+    // But we can send Map<String, String>, String, UInt8List, Uri.
+    final requestBody = await request.finalize().toBytes();
+    final requestHeaders = request.headers;
+    final requestMethod = request.method;
+    final requestUrl = request.url;
 
+    final (statusCode, reasonPhrase, responseHeaders, responseBody) =
+        await Isolate.run(() async {
       final httpUrlConnection = URL
-          .ctor3(request.url.toString().toJString())
+          .ctor3(requestUrl.toString().toJString())
           .openConnection()
           .castTo(HttpURLConnection.type, deleteOriginal: true);
 
-      request.headers.forEach((headerName, headerValue) {
+      requestHeaders.forEach((headerName, headerValue) {
```

Comment:
Prefer using a normal `for` loop. https://dart.dev/tools/linter-rules/avoid_function_literals_in_foreach_calls